exercises/dominoes/example.rb

```diff
@@ -0,0 +1,37 @@
+module Dominoes
+  def self.chain(dominoes)
+    return dominoes if dominoes.empty?
+
+    first = dominoes.first
+
+    subchain = try_subchain(dominoes.drop(1), *first)
+    subchain && [first] + subchain
+  end
+
+  def self.try_subchain(dominoes, chain_left, chain_right)
+    return chain_left == chain_right ? [] : nil if dominoes.empty?
+
+    dominoes.each_with_index { |domino, i|
+      other_dominoes = dominoes.take(i) + dominoes.drop(i + 1)
+      domino_left, domino_right = domino
+      if domino_left == chain_right
+        # Try to add domino (unflipped) to chain
+        if (subchain = try_subchain(other_dominoes, chain_left, domino_right))
+          return [domino] + subchain
+        end
+      elsif domino_right == chain_right
+        # Try to add domino (flipped) to chain
+        if (subchain = try_subchain(other_dominoes, chain_left, domino_left))
+          return [[domino_right, domino_left]] + subchain
+        end
+      end
+    }
+
+    # Found no suitable chain.
+    nil
```

Comment:
unfortunately, it's going to have to be `nil` and not `[]`.

Why? Because for the input `Dominoes.chain([])`, `[]` is a valid answer. Therefore, returning `[]` when you have no chain will mean that the answer `[]` is ambiguous.